plugins/org.jkiss.dbeaver.ext.oceanbase/src/org/jkiss/dbeaver/ext/oceanbase/model/plan/OceanbasePlanNodeJSON.java

```diff
@@ -0,0 +1,226 @@
+package org.jkiss.dbeaver.ext.oceanbase.model.plan;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.model.exec.plan.DBCPlanCostNode;
+import org.jkiss.dbeaver.model.impl.PropertyDescriptor;
+import org.jkiss.dbeaver.model.impl.plan.AbstractExecutionPlanNode;
+import org.jkiss.dbeaver.model.meta.Property;
+import org.jkiss.dbeaver.model.preferences.DBPPropertyDescriptor;
+import org.jkiss.dbeaver.model.preferences.DBPPropertySource;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.utils.CommonUtils;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+public class OceanbasePlanNodeJSON extends AbstractExecutionPlanNode implements DBCPlanCostNode, DBPPropertySource {
+
+    private OceanbasePlanNodeJSON parent;
+    private String name;
+    private JsonObject object;
+    private Map<String, String> nodeProps = new LinkedHashMap<>();
+    private List<OceanbasePlanNodeJSON> nested = new ArrayList<>();
+
+    public OceanbasePlanNodeJSON(OceanbasePlanNodeJSON parent, String name, JsonObject object) {
+        this.parent = parent;
+        this.name = name;
+        this.object = object;
+
+        parseObject(name, object);
+    }
+
+    public OceanbasePlanNodeJSON(OceanbasePlanNodeJSON parent, Map<String, String> attributes) {
+        this.parent = parent;
+        this.nodeProps.putAll(attributes);
+    }
+
+    public Map<String, String> getNodeProps() {
+        return nodeProps;
+    }
```

Comment:
Can be package-private.

It's also worth wrapping `nodeProps` in `Collections.unmodifiableMap()` for encapsulation reasons.

Reply:
If I wrap `nodeProps` into `unmodifiableMap()`, it can't be modified. But I still need to call its `put` method, so I maybe keep it.